pkg/util/hosted_zone.go

```diff
@@ -1,21 +1,21 @@
-package provisioner
+package util
 
 import (
 	"fmt"
 	"net/url"
 	"strings"
 )
 
-// getHostedZone gets derrive hosted zone from an APIServerURL.
-func getHostedZone(APIServerURL string) (string, error) {
-	url, err := url.Parse(APIServerURL)
+// GetHostedZone gets the hosted zone from an APIServerURL.
+func GetHostedZone(apiServerURL string) (string, error) {
```

Comment:
Since we promote this into util and make exported maybe it makes sense to pass cluster object (or make it a method of cluster) - this way in the future we may use e.g. dedicated cluster field or config item to store hosted zone.

Reply:
Good idea! But I would like to keep it out of this PR, hoping no one relies on the function outside of the repo. I was merely moving it around to avoid duplication. It's all quite hacky in first place 😕